tests/e2e/vmlistener_test.go

```diff
@@ -186,6 +192,102 @@ func (s *VMListenerTestParams) TestFailoverAcrossVM(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// 1. Restart hostd
+	out, err := ssh.InvokeCommand(s.esx, restartHostd)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Give hostd time to start up
+	misc.SleepForSec(30)
```

Comment:
Actually sleeping for a fixed amount of time I believe is the most robust way. Based on review comments i'd added a polling mechanism for docker restart in #1500 and with that I'm seeing very inconsistent results from the tests.

Twice it has failed because the refcounting hasn't completed in the plugin post docker restart (plugin also restarts). And volume doesn't get detached, causing test failure. Overall, these PRs are taking like a couple of days max to code and test and then a couple of weeks to get CI to pass.

The good part is I found two bugs from these failed CIs and one is fixed.

Being exact via a polling mechanism only tells that hostd is started not whether hostd is ready to start accepting requests. If hostd isn't ready thats something the ESX service needs to take care of.